packages/yoastseo/src/languageProcessing/researches/getSubheadingTextLengths.js

```diff
@@ -15,15 +15,19 @@ export default function( paper, researcher ) {
 	const text = excludeTableOfContentsTag( paper.getText() );
 	const matches = getSubheadingTexts( text );
 
+	//return an array of objecs with subheading and text
+
 	// An optional custom helper to count length to use instead of countWords.
 	const customCountLength = researcher.getHelper( "customCountLength" );
 
 	const subHeadingTexts = [];
-	forEach( matches, function( subHeading ) {
+	forEach( matches.foundSubheadings, function( match) {
 		subHeadingTexts.push( {
-			text: subHeading,
-			countLength: customCountLength ? customCountLength( subHeading ) : countWords( subHeading ),
+			subheading: match.subheading,
+			text: match.text,
+			countLength: customCountLength ? customCountLength( match.text ) : countWords( match.text ),
+			index: match.index,
 		} );
 	} );
-	return subHeadingTexts;
+	return { subHeadingTexts: subHeadingTexts, text: matches.text };
```

Comment:
Since what is returned has changed, the JSDoc of this function should be updated